pages/bookshelf/series/_id.vue

```diff
@@ -19,10 +23,167 @@ export default {
+
+      const payload = await this.$nativeHttp.get(`/api/libraries/${this.series.libraryId}/${entityPath}${fullQueryString}`).catch((error) => {
+        console.error('failed to fetch books', error)
+        return null
+      })
+
+      if (payload && payload.results) {
+        console.log('Received payload', payload)
+        this.books = payload.total
+
+        for (let i = 0; i < payload.results.length; i++) {
+          console.log(payload.results[i].numFiles)
+          if (!(await this.$db.getLocalLibraryItem(`local_${payload.results[i].id}`))) {
+            this.numFiles += payload.results[i].numFiles
+            this.numAudioFiles += payload.results[i].media.numAudioFiles
+            this.libraryIds.push(payload.results[i].id)
+          }
+        }
+      }
+      let totalPages = Math.ceil(this.books / this.booksPerFetch)
+      if (totalPages > page + 1) {
+        return false
+      }
+      return true
+    },
+    async download(selectedLocalFolder = null) {
+      // Get the local folder to download to
+      let localFolder = selectedLocalFolder
+      if (!this.isIos && !localFolder) {
+        const localFolders = (await this.$db.getLocalFolders()) || []
+        console.log('Local folders loaded', localFolders.length)
+        const foldersWithMediaType = localFolders.filter((lf) => {
+          console.log('Checking local folder', lf.mediaType)
+          return lf.mediaType == this.mediaType
+        })
+        console.log('Folders with media type', this.mediaType, foldersWithMediaType.length)
+        const internalStorageFolder = foldersWithMediaType.find((f) => f.id === `internal-${this.mediaType}`)
+        if (!foldersWithMediaType.length) {
+          localFolder = {
+            id: `internal-${this.mediaType}`,
+            name: this.$strings.LabelInternalAppStorage,
+            mediaType: this.mediaType
+          }
+        } else if (foldersWithMediaType.length === 1 && internalStorageFolder) {
+          localFolder = internalStorageFolder
+        } else {
+          this.$store.commit('globals/showSelectLocalFolderModal', {
+            mediaType: this.mediaType,
+            callback: (folder) => {
+              this.download(folder)
+            }
+          })
+          return
+        }
+      }
+
+      // Fetch serie data from server
+      let page = 0
+      let fetchFinished = false
+      while (fetchFinished === false) {
+        fetchFinished = await this.fetchSerieEntities(page)
+        page += 1
+      }
+      if (fetchFinished !== true) {
+        console.error('failed to fetch serie books data')
+        return null
+      }
+
+      // Format message for dialog
+      let startDownloadMessage = `Serie "${this.series.name}" has ${this.books} book${this.books == 1 ? '' : 's'}. Download missing ${this.libraryIds.length} book${this.libraryIds.length == 1 ? '' : 's'} with ${this.numFiles} file${this.numFiles == 1 ? '' : 's'}`
```

Comment:
We can make these strings translatable by setting them up like the following in `strings/en-us.json`.
`SeriesDownloadMessage: "Download missing {0} book(s) with {1} file(s)?",`
We can then use the following to substitute values into the string for display. The `{num}` syntax is replaced by the array index below.
`return this.$getString('SeriesDownloadMessage', [this.libraryIds.length, this.numFiles])`

I think we can also simplify the string to just be what is going to be downloaded, the user doesn't need to see how many things are already downloaded.

Reply:
Maybe the string should instead just be "download missing {0} book(s) of {1} file(s), totaling {2} MB?". That way we can simplify the messages some more and don't need to worry about identifying ebook versus other file types, and users probably want to know how big of a download it would be.

Reply:
Changed download message to suggested.